assets/js/components/surveys/SurveyQuestionMultiSelect.js

```diff
@@ -0,0 +1,199 @@
+	const mappedChoices = choices.map( ( { answer_ordinal, write_in } ) => {
+		const optionalKeys = write_in ? { answer_text: '' } : {};
+
+		return {
+			answer_ordinal,
+			selected: false,
+			...optionalKeys,
+		};
+	} );
+
+	const initialState = keyBy( mappedChoices, 'answer_ordinal' );
+
+	const [ selectedValues, setSelectedValues ] = useState( initialState );
+
+	const handleCheck = ( answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				selected: ! selectedValues[ answer_ordinal ].selected,
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	const handleAnswerChange = ( event, answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				answer_text: event.target.value?.slice(
+					0,
+					MAXIMUM_CHARACTER_LIMIT
+				),
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	const handleSubmit = () => {
+		const answer = Object.values( selectedValues )
+			.filter( ( { selected } ) => selected )
+			.map( ( { answer_ordinal, answer_text } ) => {
+				if ( answer_text ) {
+					return { answer_ordinal, answer_text };
+				}
+				return { answer_ordinal };
+			} );
+		answerQuestion( {
+			answer,
+		} );
+	};
+
+	const hasEmptySelectedTextValue =
+		choices.filter( ( { write_in, answer_ordinal } ) => {
+			if ( write_in ) {
+				const { selected, answer_text } = selectedValues[
+					answer_ordinal
+				];
+				if ( selected && answer_text.length === 0 ) {
+					return true;
+				}
+			}
+			return false;
+		} ).length > 0;
+
+	const totalSelectedValues = Object.values( selectedValues ).filter(
+		( { selected } ) => selected
+	).length;
+	const hasLessThanMinChoices = totalSelectedValues < min_choices;
+	const hasMoreThanMaxChoices =
+		max_choices && totalSelectedValues > max_choices;
+
+	const isSubmitButtonDisabled =
+		hasEmptySelectedTextValue ||
+		hasLessThanMinChoices ||
+		hasMoreThanMaxChoices;
+
+	return (
+		<div className="googlesitekit-survey__multi-select">
+			<SurveyHeader title={ question } dismissSurvey={ dismissSurvey } />
+
+			<div className="googlesitekit-survey__body">
+				{ choices.map( ( { answer_ordinal, text, write_in } ) => {
+					const answer = selectedValues[ answer_ordinal ];
+
+					return (
+						<div key={ text } style={ { display: 'flex' } }>
+							<Checkbox
+								checked={ answer.selected }
+								onChange={ () => handleCheck( answer_ordinal ) }
+								value={ `${ answer_ordinal }` }
+								id={ text }
+								name={ text }
+							>
+								{ text }
+							</Checkbox>
+							{ write_in && (
+								<TextField>
+									<Input
+										onChange={ ( event ) =>
+											handleAnswerChange(
+												event,
+												answer_ordinal
+											)
+										}
+										value={ answer.answer_text }
+										disabled={ ! answer.selected }
+										aria-label={ `Text input for option ${ text }` }
```

Comment:
Why can't we use a regular label here? It could be wrapped in `VisuallyHidden` if needed, but a normal label tag here seems fine.

Reply:
This should be localised as well, eg. `sprintf( __( 'Text input for option %s', 'google-site-kit' ), text )`

Reply:
I've added a label `VisuallyHidden`. I wasn't aware that there was a difference to screen readers

Reply:
Screen readers _will_ use the `aria-label`, but the label tag is arguably semantically a bit more correct and easier for things to parse. It's the better default is all 🙂